src/SourceBuild/content/eng/finish-source-only.proj

```diff
@@ -131,8 +131,12 @@
       <!-- Exclude the Private.SourceBuilt.Artifacts archive from poison usage scan. -->
       <PoisonFileToCheck Include="$(ArtifactsAssetsDir)*$(ArchiveExtension)" />
       <PoisonFileToCheck Remove="$(ArtifactsAssetsDir)$(SourceBuiltArtifactsTarballName)*" />
+
       <!-- Include shipping nuget packages. -->
-      <PoisonFileToCheck Include="$(ArtifactsShippingPackagesDir)*.nupkg" />
+      <ShippingPackageToCheck Include="$(ArtifactsShippingPackagesDir)**/*.nupkg" />
+      <Error Condition="'@(ShippingPackageToCheck)' == ''" Text="No shipping packages will be poison checked - this is unexpected!" />
```

Comment:
Add another similar check for `PoisonFileToCheck`?

Reply:
Sure, good idea - thanks!

Reply:
Checking at the end would never work as the new check I'm adding for shipping packages would fail first. But, I can add a check for assets, i.e. the value of `PoisonFileToCheck`, right before adding shipping packages.

Reply:
Fixed with https://github.com/dotnet/sdk/pull/47176/commits/deb765a4f9661e93f9de5fb0eabc055a31042e04

I've added a check for assets.

Reply:
Actually, can we divide these categories of files into separate item groups that we can then validate individually? For example, defined an item group for the PSB artifacts and a separate one for the SBRPs. Check that each of those item groups are non-empty, just as you do for the separate item group of the shipping packages. Then at the end, add them all to `PoisonFileToCheck`.

Reply:
I was also thinking of that and it does make sense. Best time to do it is now 👍

Reply:
Fixed with https://github.com/dotnet/sdk/pull/47176/commits/5d800782d5a380f9d67238c66a16dc60c43b5356